File: 03.Custom View/MyCustomView/app/src/main/java/com/dicoding/picodiploma/mycustomview/MainActivity.java
package com.dicoding.picodiploma.mycustomview;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private MyButton myButton;
    private MyEditText myEditText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        myButton = findViewById(R.id.my_button);
        myEditText = findViewById(R.id.my_edit_text);

        // Melakukan pengecekan saat pertama kali activity terbentuk
        if (myEditText.getText()!=null){
            setMyButtonEnable(myEditText.getText().toString());
        }

        // Menambahkan metode ketika text terjadi perubahan
        myEditText.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                setMyButtonEnable(s.toString());
            }

            @Override
            public void afterTextChanged(Editable s) {

            }
        });

        // Menambahkan aksi klik kepada button
        myButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this, myEditText.getText().toString(), Toast.LENGTH_SHORT).show();
            }
        });
    }

    // Metode untuk mengubah disable dan enable pada button
    private void setMyButtonEnable(String string){
        if (string.isEmpty()){
            myButton.setEnabled(false);
        }else{
            myButton.setEnabled(true);
        }
    }
}

File: Custom View/MyCustomView/kotlin/app/src/main/java/com/dicoding/picodiploma/mycustomview/MainActivity.kt
package com.dicoding.picodiploma.mycustomview

import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Melakukan pengecekan saat pertama kali activity terbentuk
        setMyButtonEnable()

        // Menambahkan metode ketika text terjadi perubahan
        my_edit_text.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {

            }

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
                setMyButtonEnable()
            }

            override fun afterTextChanged(s: Editable) {

            }
        })

        // Menambahkan aksi klik kepada button
        my_button.setOnClickListener { Toast.makeText(this@MainActivity, my_edit_text.text, Toast.LENGTH_SHORT).show() }
    }

    // Metode untuk mengubah disable dan enable pada button
    private fun setMyButtonEnable() {
        val result = my_edit_text.text
        my_button.isEnabled = result != null && result.toString().isNotEmpty()
    }
}
